support-utils/cre_astro_software_gather_logs.sh

```diff
@@ -0,0 +1,392 @@
+    PVC_DIR="$NS/PVC"
+    mkdir -p "$PVC_DIR"
+
+    # Loop through each PVC and describe it
+    for pvc in $pvcs; do
+      echo "Describing PVC '$pvc' in namespace '$NS'"
+      kubectl describe pvc "$pvc" -n "$NS" > "$PVC_DIR/${pvc}_describe.txt"
+      echo "PVC '$pvc' described and saved to $PVC_DIR/${pvc}_describe.txt"
+    done
+  else
+    echo "No PVCs found in namespace '$NS'."
+  fi
+
+  echo "======================Gathering all Pods status in $NS======================"
+  # Get status of all pods in the namespace
+  kubectl get pods -n "$NS" -o wide > "$NS/${NS}_all_pods_status.txt"
+
+  echo "======================Gathering Describe output of all pods in $NS======================"
+  # Loop through all pods and describe them
+  for POD in $(kubectl get pods --no-headers -n "$NS" | awk '{print $1}'); do
+    echo "Describing pod '$POD' in namespace '$NS'"
+    kubectl describe pod "$POD" -n "$NS" > "$NS/DescribePods/${POD}_describe.txt"
+  done
+
+  echo "======================Gathering Describe output of Bad state pod in $NS======================"
+  # Loop through pods in a bad state
+  for BAD_POD in $(kubectl get pods --no-headers -n "$NS" | grep -v Running | grep -v Completed | awk '{print $1}'); do
+    echo "$BAD_POD pod is in a bad state"
+    echo "Collecting Describe output of bad state pod $BAD_POD in $NS Namespace"
+    kubectl describe pod "$BAD_POD" -n "$NS" > "$NS/BadPods/$BAD_POD-BAD_$NS.describe.txt"
+  done
+
+  echo "======================Gathering log output of Bad state pod in $NS======================"
+  # Gather logs of pods in a bad state
+  for BAD_POD in $(kubectl get pods --no-headers -n "$NS" | grep -v Running | grep -v Completed | awk '{print $1}'); do
+    echo "Starting to collect log of the bad state pod $BAD_POD in $NS namespace"
+    for container_name in $(kubectl get pods "$BAD_POD" -o jsonpath='{.spec.containers[*].name}' -n "$NS" | awk '{NF-=0; OFS="\n"; $1=$1}1' | sort); do
+      echo "Collecting log of the container $container_name in pod $BAD_POD in the $NS namespace now"
+      kubectl logs "$BAD_POD" -n "$NS" -c "$container_name" > "$NS/BadPods/$BAD_POD-pod_$container_name-BADPODLOG.log"
+    done
+  done
+
+  echo "======================Gathering logs of All the pods in $NS======================"
+  # Gather logs of all pods
+  for POD in $(kubectl get pods --no-headers -n "$NS" | awk '{print $1}'); do
+    echo "Starting to collect log of the pod $POD in $NS namespace"
+    for container_name in $(kubectl get pods "$POD" -o jsonpath='{.spec.containers[*].name}' -n "$NS" | awk '{NF-=0; OFS="\n"; $1=$1}1' | sort); do
+      echo "Collecting log of the container $container_name in pod $POD in the $NS namespace now"
+      kubectl logs "$POD" -n "$NS" -c "$container_name" > "$NS/AllPodlogs/$POD-pod_$container_name-container.log"
+    done
+  done
+
+  echo "======================Gathering Events in $NS======================"
```

Comment:
We should probably replace all these `echo "======= whatever ======"` with a function that automatically does the formatting.